packages/features/ee/impersonation/lib/ImpersonationProvider.ts

```diff
@@ -85,6 +86,26 @@ const ImpersonationProvider = CredentialsProvider({
     checkUserIdentifier(creds);
     checkPermission(session);
 
```

Comment:
In ImpersonationProvider I don't see any  error thrown  if `session` is undefined but the typing says it can be ? can it actually happen a user reaches this point without having a session?

Reply:
It would get caught down the trea when looking for the session user but yeah deffo worth adding an a guard